examples/vendor/README.md

```diff
@@ -0,0 +1,17 @@
+# Vendor
+
+Vendors wanting to add custom CLI commands to `otbr-agent` may implement a `ot-vendor-cli` library.
```

Comment:
Perhaps reword to something like "Vendors wanting to add specialized features, such as custom CLI commands, to `otbr-agent` may implement a `ot-vendor` library."